Adding support for V2 shift profile

fiveohhh opened the pull request:
Adds support for the newishly released v2 shift profile
https://www.thisisant.com/developer/resources/tech-bulletin/ant-shifting-2-profile-now-available-to-ant-members

tuna-f1sh commented on openant/devices/shift.py:
```diff
@@ -43,6 +66,25 @@ class ShiftData(DeviceData):
     invalid_outboard_front: int = field(default=0)
     shift_failure_rear: int = field(default=0)
     shift_failure_front: int = field(default=0)
+    function_set_1_configuration: FunctionSetConfiguration = field(default=FunctionSetConfiguration())
```
The tests are failing because a mutable default needs to use `default_factory` here (and all `FunctionSetConfiguration()`.

fiveohhh replied:
Updated to use default_factory

tuna-f1sh commented on openant/devices/shift.py:
```diff
@@ -99,10 +140,60 @@ def on_data(self, data):
+
+            self.data["shift"].event_count = data[1]
+
+            self.data["shift"].event_1 = FunctionSetEvent(data[2] & 0x0F, (data[2] & 0x03) >> 4)
```
The second parameter needs to be wrapped with `FunctionSetEventType` to cast as the Enum. Even with the typing Python doesn't auto cast so this will be a int in the dataclass currently.

fiveohhh replied:
Now correctly cast

tuna-f1sh commented on openant/devices/shift.py:
```diff
@@ -99,10 +140,60 @@ def on_data(self, data):
+                _logger.info(f"Shifting status update {self}: {self.data['shift']}")
+                self.on_device_data(page, "shift_system_status", self.data["shift"])
+        elif page == 0x03:
+            self.data["shift"].function_set_1_configuration = FunctionSetConfiguration(data[1] & 0x02, data[1] & 0x04, data[1] & 0x08)
```
Same as above, these will be `int` not `bool`. They need casting to `bool` with a operator like `data[1] & 0x02 == 0x02`.

tuna-f1sh commented on openant/devices/shift.py:
```diff
@@ -28,6 +28,29 @@ class ShiftingSystemID(Enum):
     def _missing_(cls, _):
         return ShiftingSystemID.Unknown
 
+class FunctionSetEventTypes(Enum):
```
Pedantic but can you rename to `FunctionSetEventType` since that's how it's used (singular).

tuna-f1sh reviewed (changes requested):
I've provided some comments of things that need fixing - it currently fails to import. Have you been able to test this with a real-world device or is it just fleshing out the spec?

fiveohhh commented:
> I've provided some comments of things that need fixing - it currently fails to import. Have you been able to test this with a real-world device or is it just fleshing out the spec?

Addressed the feedback.  Yes, I tested this on a SRAM mountain RD and got the results I was expecting.

tuna-f1sh commented:
Great thanks!
